### PR TITLE
Now properly handling checks

### LINKED ISSUE #105: checkGit into abstract function
As a developer, I do not want to make a separate for every single thing.
As a developer, I want to follow GoF practises on software design.
As a user, I want that the software I'm using is efficient and takes less space.

### PR DESCRIPTION
**Pull request by veikkahonkanen**
Resolves #105

### CODE REVIEW

**Review by eetupur — APPROVED** [review]

**Review by eetupur — APPROVED** [review]

File: src/classes/check-utils.ts [hunk 1]
```diff
@@ -12,84 +12,39 @@ export class CheckUtils {
    * @param toCheck contains a CheckSet array with prerequisites to check
    */
   static checkPreq = async (toCheck: CheckSet[]): Promise<CheckErrorSet[]> => {
-    const erroList: CheckErrorSet[] = [];
+    const erroList : CheckErrorSet[] = [];
```
Comment (jarinykanen) [inline src/classes/check-utils.ts]:
```suggestion
    const erroList: CheckErrorSet[] = [];
```

File: src/classes/check-utils.ts [hunk 2]
```diff
@@ -12,84 +12,39 @@ export class CheckUtils {
...
 
     for (const currentCheck of toCheck) {
       try {
-        const checkResult: CheckErrorSet = await checkRouter(currentCheck);
+        const checkResult : CheckErrorSet = await checkHandler(currentCheck);
```
Comment (jarinykanen) [inline src/classes/check-utils.ts]:
```suggestion
        const checkResult: CheckErrorSet = await checkHandler(currentCheck);
```

File: src/classes/check-utils.ts [hunk 3]
```diff
@@ -12,84 +12,39 @@ export class CheckUtils {
...
+    const installed: boolean = await InstallUtils.isInstalled(currentCheck.checkable);
 
-      case Software.GitHub:
-        //To do: add check
-        console.log(`Should add check for ${Software.GitHub}`);
-        return await checkGit();
+    return {
+      check : currentCheck.checkable,
+      error : installed ? false : true,
+      details : installed ? null : `${currentCheck.checkable} not installed!`
```
Comment (jarinykanen) [inline src/classes/check-utils.ts]:
```suggestion
      check: currentCheck.checkable,
      error: installed ? false : true,
      details: installed ? null : `${currentCheck.checkable} not installed!`
```

**Review by jarinykanen — CHANGES_REQUESTED** [review]

File: src/classes/check-utils.ts [hunk 4]
```diff
@@ -16,81 +16,36 @@ export class CheckUtils {
...
 /**
- * Routes details to relevant check functions
+ * Handles runned checks one at a time
  * 
- * @param currentCheck details of current checkable command
+ * @param {CheckSet} currentCheck  details of current checkable command
+ * @returns {CheckErrorSet} that contains software name, error and possible details
  */
-const checkRouter = async (currentCheck: CheckSet) => {
+const checkHandler = async (currentCheck: CheckSet) => {
```
Comment (jarinykanen) [inline src/classes/check-utils.ts]: Add return type to this function

**Review by jarinykanen — CHANGES_REQUESTED** [review]

File: src/classes/check-utils.ts (hunk 4 above)
Comment (veikkahonkanen) [reply src/classes/check-utils.ts]: Added return type

**Pull request merged**